File: etc/disks/server-echo.js
const WebSocket = require('ws');

const wss = new WebSocket.Server({ port: 8080 });

	
var connections = {}

function sendAll( group ){
    var wsarray = connections[group];
    var wsarray2 = [];
    var x = JSON.stringify( data);
    for (var i=0; i< wsarray.length; i++){
        var ws = wsarray[i];
        try{
           ws.send( x );
            //console.log( x );
        }catch(e){continue;}
        wsarray2.push(ws);    
    }
    connections[group] = wsarray2;
}

wss.on('connection', function connection(ws) {
    ws.on('message', function incoming(message) {
        data = JSON.parse( message);
        group = data.group;
        wsarray = connections[group];
        if (! wsarray){
            wsarray = connections[group] = [];
        }
        if (wsarray.indexOf(ws) <0){
            wsarray.push(ws);
        }
        sendAll( group );
     });  
});

File: etc/disks/server-echo.py
import asyncio
import websockets
import json

connections = {}

async def hello(websocket, path):
    while True:
        try:
            data = await websocket.recv()            

            group = json.loads( data)['group']
            
            if group not in connections:
                connections[group] = set()
            wsarray = connections[group]
            if websocket not in wsarray:
                wsarray.add( websocket)

            wsarray2 =  set()
            #print(1)
            for ws in wsarray:
               try:
                    await ws.send(data)
                    #print(2)
               except Exception as e:
                    print( e)
                    continue
               wsarray2.add(ws)
            connections[group] = wsarray2
        except Exception as e:
            print( e)
            return        

start_server = websockets.serve(hello, "192.168.1.106", 8080)

asyncio.get_event_loop().run_until_complete(start_server)
asyncio.get_event_loop().run_forever()
